pages/api/favorites/[episodeNumber].js

```diff
@@ -2,34 +2,37 @@ import dbConnect from "@/db/connect";
 import User from "@/db/models/User";
 import { getToken } from "next-auth/jwt";
 
-export default async function handler(req, resp) {
+export default async function handler(req, res) {
   await dbConnect();
   const token = await getToken({ req });
   const { episodeNumber } = req.query;
 
-  if (token && req.method === "PUT") {
-    const updatedUser = await User.findOneAndUpdate(
-      { sub: token.sub },
-      { $push: { favorites: parseInt(episodeNumber) } }
-    );
+  if (token)
+    switch (req.method) {
```

Comment:
since you are refactoring your endpoints, this would be a good time to have a default error response for unmapped requests.

if you don't have a default response, there is a risk of requests not finishing with a response, but ending in a timeout or broken connection instead.

the unmapped default response should be a [status code 405](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) with an allow header, that specifies which methods are mapped.

Reply:
I implemented it, hope its correct.